openedx/core/djangoapps/courseware_api/views.py

```diff
@@ -268,3 +273,61 @@ def get(self, request, usage_key_string, *args, **kwargs):
             str(usage_key),
             disable_staff_debug_info=True)
         return Response(json.loads(sequence.handle_ajax('metadata', None)))
+
+
+class Resume(DeveloperErrorViewMixin, APIView):
+    """
+    **Use Cases**
+
+        Request the last completed block in a course
+
+    **Example Requests**
+
+        GET /api/courseware/resume/{course_key}
+
+    **Response Values**
+
+        Body consists of the following fields:
+
+            * block: the last completed block key
+            * section: the key to the section
+            * unit: the key to the unit
+        If no completion data is available, the keys will be null
+
+    **Returns**
+
+        * 200 on success with above fields.
+        * 400 if an invalid parameter was sent.
+        * 403 if a user who does not have permission to masquerade as
+          another user specifies a username other than their own.
+        * 404 if the course is not available or cannot be seen.
+    """
+
+    authentication_classes = (
+        JwtAuthentication,
+        SessionAuthenticationAllowInactiveUser,
+    )
+    permission_classes = (IsAuthenticated, )
+
+    def get(self, request, course_key_string, *args, **kwargs):
+        """
+        Return response to a GET request.
+        """
+        course_id = CourseKey.from_string(course_key_string)
+        resp = {
+            'block': None,
+            'section': None,
+            'unit': None,
+        }
+
+        try:
+            block_key = get_key_to_last_completed_block(request.user, course_id)
```

Comment:
So is this is a really slow function or something?  Curious why we need to make this its own API endpoint, rather than adding it into the existing course metadata endpoint...

Reply:
Since the information here is only used if the course is requested without a sequence, and it has to hit the modulestore in order to find the path to the block, I think this should be it's own request.